FcmJava/fcmjava-client/src/main/java/io/github/shsmysore/fcmjava/client/http/apache/DefaultHttpClient.java

```diff
@@ -104,24 +104,47 @@ public DefaultHttpClient(IFcmClientSettings settings,
         }
     }
 
-    private <T> CompletableFuture<HttpResponse<T>> tryResend(HttpClient client,
-                                                                   HttpRequest request,
-                                                                   HttpResponse.BodyHandler<T> handler,
-                                                                   int count,
-                                                                   HttpResponse<T> resp) {
-        if (resp.statusCode() < 500 || count > MAX_RETRY) {
+    private <T> CompletableFuture<HttpResponse<T>> retrySend(HttpClient client,
+                                                             HttpRequest request,
+                                                             HttpResponse.BodyHandler<T> handler,
+                                                             int count,
+                                                              HttpResponse<T> resp) {
+        if ((resp != null && resp.statusCode() < 500) || count > MAX_RETRY) {
+            return toResponse(resp);
+        } else {
+            return client.sendAsync(request, handler)
+                    .thenApply(Optional::of)
+                    .exceptionally(ex -> {
+                        // ToDo: Only retry on IOException. Others should be thrown again.
```

Comment:
Just logging in for now during my testing. I'll have a follow-up commit to addressing it.